Bug:  Array variable accepts non-array value

## Issue Description

Array variables can be assigned non-array values without any error. Array values should be required to be encapsulated in `{}`, but currently the type checker allows assigning primitive values directly to array-typed variables.

## Example Code

```ez
do main(){
    const arr [int] = 10 //THIS SHOULD NOT BE ALLOWED
    println(arr) //output: 10
}
```

## Expected Behavior

The code should produce a type error:

```
error[E2001]: type mismatch in variable declaration
  --> main.ez:2:23
   |
2 |     const arr [int] = 10
   |                       ^^ expected array type [int], found INTEGER
   |

Help: Array values must be encapsulated in curly braces {}
      Example: const arr [int] = {10}
```

## Current Behavior

The code runs without error and outputs: `10`

The variable `arr` is declared as type `[int]` (array of integers) but is being assigned the integer value `10` directly, which should not be allowed.

## Root Cause

The type checker is not validating that array-typed variables receive array values. It appears to be accepting any value that matches the element type instead of requiring the full array type.

## Implementation Details

**Location:** `pkg/interpreter/evaluator.go` - variable declaration evaluation

The type checking logic needs to:
1. Verify that when a variable is declared with an array type `[T]`, the assigned value must be of type `ARRAY_OBJ`
2. Additionally verify that the array's element types match the declared element type
3. Reject assignments where a primitive/non-array value is assigned to an array-typed variable

**Suggested Fix:**
In the constant/variable declaration evaluation:
- Check if the declared type is an array type (contains `[` and `]`)
- If so, verify the assigned value is an `*Array` object
- Provide clear error message if type mismatch occurs

## Related

This is a type safety issue that could lead to unexpected runtime behavior and bugs in user code.

## Priority

Medium - This is a type safety violation but may not cause crashes in typical usage

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -376,9 +376,22 @@ func evalVariableDeclaration(node *ast.VariableDeclaration, env *Environment) Ob
 			return val
 		}
 
-		// If a type is declared and we have an integer value, set the declared type
-		// and validate signed/unsigned compatibility
+		// Validate type compatibility if a type is declared
 		if node.TypeName != "" {
+			// Check if declared type is an array type
+			if len(node.TypeName) > 0 && node.TypeName[0] == '[' {
+				// Array type declared - value must be an array
+				if _, ok := val.(*Array); !ok {
+					return newErrorWithLocation("E3018", node.Token.Line, node.Token.Column,
+						"type mismatch: expected array type '%s', got %s\n\n"+
+							"Array values must be enclosed in curly braces {}\n"+
+							"Example: const arr %s = {%s}",
+						node.TypeName, getEZTypeName(val), node.TypeName, val.Inspect())
+				}
+			}
+
+			// If we have an integer value, set the declared type
+			// and validate signed/unsigned compatibility
 			if intVal, ok := val.(*Integer); ok {
 				// Check for negative value assigned to unsigned type
 				if isUnsignedIntegerType(node.TypeName) && intVal.Value < 0 {

--- a/test/array_type_validation_test.ez
+++ b/test/array_type_validation_test.ez
@@ -1,0 +1,136 @@
+/*
+ * EZ Language - Array Type Validation Tests
+ *
+ * This file tests that array type validation works correctly:
+ * - Array-typed variables must receive array values
+ * - Valid array declarations should work
+ * - Dynamic and fixed-size arrays should work
+ */
+
+import @std
+import @arrays
+
+do main() {
+    using std
+
+    println("========================================")
+    println("  Array Type Validation Tests          ")
+    println("========================================")
+    println("")
+
+    test_valid_array_declarations()
+    test_empty_arrays()
+    test_fixed_size_arrays()
+    test_array_operations()
+
+    println("")
+    println("========================================")
+    println("  All Array Validation Tests Passed!   ")
+    println("========================================")
+}
+
+// ============================================================================
+// TEST: Valid Array Declarations
+// ============================================================================
+
+do test_valid_array_declarations() {
+    using std
+    println("-> Testing valid array declarations...")
+
+    // Integer arrays
+    temp int_arr [int] = {1, 2, 3, 4, 5}
+    const const_int_arr [int] = {10, 20, 30}
+
+    // String arrays
+    temp str_arr [string] = {"hello", "world"}
+    const const_str_arr [string] = {"a", "b", "c"}
+
+    // Boolean arrays
+    temp bool_arr [bool] = {true, false, true}
+
+    // Float arrays
+    temp float_arr [float] = {1.5, 2.5, 3.5}
+
+    // Character arrays
+    temp char_arr [char] = {'a', 'b', 'c'}
+
+    // Verify lengths
+    println("  int_arr length: ${len(int_arr)}")
+    println("  str_arr length: ${len(str_arr)}")
+    println("  bool_arr length: ${len(bool_arr)}")
+
+    println("  [OK] Valid array declarations work")
+}
+
+// ============================================================================
+// TEST: Empty Arrays
+// ============================================================================
+
+do test_empty_arrays() {
+    using std
+    println("-> Testing empty arrays...")
+
+    // Empty arrays with explicit empty braces
+    temp empty_int [int] = {}
+    temp empty_str [string] = {}
+
+    // Empty arrays without initialization (default)
+    temp default_arr [int]
+
+    println("  empty_int length: ${len(empty_int)}")
+    println("  empty_str length: ${len(empty_str)}")
+    println("  default_arr length: ${len(default_arr)}")
+
+    println("  [OK] Empty arrays work")
+}
+
+// ============================================================================
+// TEST: Fixed-Size Arrays
+// ============================================================================
+
+do test_fixed_size_arrays() {
+    using std
+    println("-> Testing fixed-size arrays...")
+
+    // Fixed-size arrays
+    const fixed_int [int, 3] = {1, 2, 3}
+    const fixed_str [string, 2] = {"hello", "world"}
+
+    println("  fixed_int length: ${len(fixed_int)}")
+    println("  fixed_str length: ${len(fixed_str)}")
+
+    // Access elements
+    println("  fixed_int[0]: ${fixed_int[0]}")
+    println("  fixed_str[1]: ${fixed_str[1]}")
+
+    println("  [OK] Fixed-size arrays work")
+}
+
+// ============================================================================
+// TEST: Array Operations
+// ============================================================================
+
+do test_array_operations() {
+    using std
+    println("-> Testing array operations...")
+
+    temp nums [int] = {1, 2, 3}
+
+    // Push
+    arrays.push(nums, 4)
+    println("  After push(4): length = ${len(nums)}")
+
+    // Pop
+    temp popped int = arrays.pop(nums)
+    println("  Popped value: ${popped}")
+
+    // Indexing
+    temp first int = nums[0]
+    println("  First element: ${first}")
+
+    // Modification
+    nums[0] = 100
+    println("  After nums[0] = 100: ${nums[0]}")
+
+    println("  [OK] Array operations work")
+}

--- a/test/test_array_type_validation.ez
+++ b/test/test_array_type_validation.ez
@@ -1,0 +1,12 @@
+/*
+ * TEST: Array type validation - non-array value should ERROR
+ * Expected: E3018 - array type requires array literal
+ */
+
+import @std
+
+do main() {
+    // This should produce error E3018
+    const arr [int] = 10
+    std.println(arr)
+}

